content/en/docs/monolake/Architecture/context.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Context Management"
+linkTitle: "Context"
+weight: 4
+---
+
+# `certain_map`
+
+In a service-oriented architecture, managing the context data that flows between different services is a critical aspect of the system design. The [`certain_map`](https://docs.rs/certain-map/latest/certain_map/) crate provides a powerful way to define and work with typed context data, ensuring the existence of required information at compile-time.
+
+## The Problem `certain_map` Solves
+
+When building modular services, it's common to have indirect data dependencies between components. For example, a downstream service may require information that was originally provided in an upstream request, but the intermediate services don't directly use that data. Traditionally, this would involve passing all potentially relevant data through the request/response types, which can quickly become unwieldy and error-prone.
+
+Alternatively, you might use a `HashMap` to manage the context data, but this approach has a significant drawback: you cannot ensure at compile-time that the required key-value pairs have been set when the data is read. This can lead to unnecessary error handling branches or even panics in your program.
+
+## How `certain_map` Helps
+
+The `certain_map` crate solves this problem by providing a typed map that ensures the existence of specific items at compile-time. When you define a `Context` struct using `certain_map`, the compiler will enforce that certain fields are present, preventing runtime errors and simplifying the implementation of your services.
```

Comment:
`a typed map`, to be more precise it is not a typed map but just for users it's like that. We can describe it as `a typed-map-like struct`